api/nodes/serializers.py

```diff
@@ -124,9 +128,126 @@ def update(self, instance, validated_data):
+    fullname = ser.CharField(read_only=True, help_text='Display name used in the general user interface')
+    given_name = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    middle_name = ser.CharField(read_only=True, source='middle_names', help_text='For bibliographic citations')
+    family_name = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    suffix = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    date_registered = ser.DateTimeField(read_only=True)
+    gravatar_url = ser.CharField(read_only=True, help_text='URL for the icon used to identify the user. '
+                                                           'Relies on http://gravatar.com ')
+    social_accounts = ser.DictField(read_only=True, source='social',
+                                    help_text='A dictionary of various social media account dentifiers including '
+                                              'an array of user-defined URLs')
+    bibliographic = ser.BooleanField(help_text='Whether the user will be included in citations for this node or not.  '
+                                               'Required due to issues with field defaulting to false.')
+
+    permission = ser.ChoiceField(choices=[osf_permissions.READ, osf_permissions.WRITE, osf_permissions.ADMIN], required=False, allow_null=True,
+                                 help_text='Highest permission the user has.  Blank input defaults write permission if '
+                                           'user is being added and to current permission if user is being edited.')
+
+    links = LinksField({'html': 'absolute_url'})
+    nodes = JSONAPIHyperlinkedIdentityField(view_name='users:user-nodes', lookup_field='pk', lookup_url_kwarg='user_id',
+                                             link_type='related')
+    detail = JSONAPIHyperlinkedIdentityField(view_name='users:user-detail', lookup_field='pk', lookup_url_kwarg='user_id', link_type='related')
+
+    class Meta:
+        type_ = 'users'
+
+    def absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        auth = Auth(self.context['request'].user)
+        node = self.context['view'].get_node()
+        contributor = get_object_or_error(User, validated_data['_id'])
+        # Node object checks for contributor existence but can still change permissions anyway
+        if contributor in node.contributors:
+            raise exceptions.ValidationError('{} is already a validated contributor'.format(contributor.username))
+
+        bibliographic = validated_data['bibliographic']
+        permissions = self.get_permissions_list(validated_data['permission']) \
+            if 'permission' in validated_data else [osf_permissions.READ, osf_permissions.WRITE]
+        node.add_contributor(contributor=contributor, auth=auth, visible=bibliographic, permissions=permissions, save=True)
+        node.reload()
```

Comment:
This reload statement is probably not necessary. The original author seems to have been using it wherever objects are saved. I'll do a search for all other uses.

Reloading is used when a change is made outside this function/ request. (eg, in unit tests, where changes made in an API call are not available inside the function until fetched from the database again explicitly).

Sorry for all the spam as we tidy things up. Really appreciate your hard work.